### PR TITLE
Adds docs for various similar level columns

### PR DESCRIPTION
**Pull request by andrewfullard**
Also moved notation to a more accessible spot.

### :pencil: Description

**Type:** :memo: `documentation` 

Attempted to write some docs about the differences between level index, ID and number. Also moved notation to a better location. Fixes issue #258 and #364

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [x] I requested two reviewers for this pull request
- [x] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: docs/reference/level_index_id_number.rst [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Global cross-matching ID for levels across multiple DataFrames i.e. between photo
+ionization cross sections, the macro atom and electronic energy levels. 
+Computed from level index. Typically has an upper and lower form to connect levels
+together.
+
+=============
+level_number
+=============
+
+DataFrame index for each electronic energy level suitable for TARDIS use.
```
Comment (wkerzendorf) [inline docs/reference/level_index_id_number.rst:29]: Index in the order of the level energies within each species. So species Si II has `level_number` 0 to n. Often used in a multi-index with `atomic_number`, `ion_number`.

File: docs/index.rst [hunk 2]
```diff
@@ -26,6 +27,13 @@ Carsus is a package to manage atomic datasets. It can read data from a variety o
     io/vald
     io/barklem2016
 
+.. toctree::
+    :maxdepth: 2
+    :caption: Reference
+    :hidden:
+    reference/notation.rst
+    reference/level_index_id_number.rst
```
Comment (andrewfullard) [inline docs/index.rst:35]: Check why this doesn't show up

File: docs/reference/level_index_id_number.rst [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Chianti and CMFGEN levels DataFrames.
+
+For photoionization cross-sections, this is used to determine the lower and upper
+levels of cross-sections.
+
+=============
+level_id
+=============
+
+Global cross-matching ID for levels across multiple DataFrames i.e. between photo
```
Comment (wkerzendorf) [inline docs/reference/level_index_id_number.rst:20]: Global unique `level_id` across all species.

File: docs/index.rst [hunk 4]
```diff
@@ -26,6 +27,13 @@ Carsus is a package to manage atomic datasets. It can read data from a variety o
...
+
 .. toctree::
     :maxdepth: 2
     :caption: Research with Carsus
```
Comment (andrewfullard) [inline docs/index.rst:39]: change references.rst to bibliography

File: docs/index.rst (hunk 2 above)
Comment (jvshields) [reply docs/index.rst:35]: Unsure if this is the answer, but perhaps check if you need a linebreak before you specify the files.

**Review by jvshields — APPROVED** [review]

**Review by Knights-Templars — APPROVED** [review]

**Pull request merged**